crates/fluvio-version-manager/src/common/manifest.rs

```diff
@@ -55,4 +62,33 @@ mod test {
 
         assert_eq!(json, std::fs::read_to_string(manifest).unwrap());
     }
+
+    #[test]
+    fn reads_manifest_from_json() {
+        let tempdir = TempDir::new().unwrap();
+        let version_manifest = VersionManifest::new(Channel::Stable, Version::new(0, 8, 0));
+        let json = serde_json::to_string_pretty(&version_manifest).unwrap();
+        let manifest = version_manifest
+            .write(tempdir.path().to_path_buf())
+            .unwrap();
+        let read_manifest = VersionManifest::open(manifest).unwrap();
+
+        assert_eq!(json, serde_json::to_string_pretty(&read_manifest).unwrap());
+    }
+
+    #[test]
+    fn fails_to_read_manifest_from_invalid_json() {
+        let tempdir = TempDir::new().unwrap();
+        let manifest = tempdir.path().join(PACKAGE_SET_MANIFEST_FILENAME);
+
+        std::fs::write(manifest.clone(), "invalid json").unwrap();
```

Comment:
Add invalid in the test file instead of TempDir.  That way it's easier to test and see